### PR TITLE
Components: replace `TabPanel` with `Tabs` in the Font Library `Modal`

### PR DESCRIPTION
**Pull request by chad1008**
## What?
Replaces the legacy TabPanel component with the new Tabs component.

## Why?
Part of the work outlined in https://github.com/WordPress/gutenberg/issues/52997

## How?
`TabPanel` is replaced by `Tabs` and its sub-components. I've also tried to clean up the use of `name` vs `id` for tab identification. `TabPanel` used a `name` property, but `Tabs` deals exclusively with `id`s. Eliminating `name` from the code will hopefully make it easier to parse. This specifcially doesn't impact the `name` property on font collections being passed in to the library, as that _nomenclature_ still felt like it made sense.

## Testing Instructions
1. Open the site editor and the Global Styles sidebar
2. Click on **Typography**
3. Click on **Manage Fonts** to open the Font Library modal
4. Confirm all three tabs (**Library**, **Upload**, and **Install Fonts** are present). Any other tabs (if present) should represent font collections provided by your current setup.
5. Compared to `trunk`, confirm that each tab visually looks the same, that it behaves the same, and the it's displaying the correct content
6. Just for fun, modify the order of the `DEFAULT_TABS` array in `packages/edit-site/src/components/global-styles/font-library-modal/index.js` so that `installed-fonts` is no longer listed first. Reload the site editor and confirm that **Library** is still the initially selected tab even if it the first one in the list

### Testing Instructions for Keyboard
1. Open the site editor and select the Global Styles sidebar
2. Once open, press [Tab] and [Space] to reach and select the **Typography** option
3. [Tab] and [Space] again to focus and activate the **Manage Fonts** button, opening the Font Library modal
4. [Tab] twice to focus the current tab.
5. Use arrow keys to navigate between tabs. Focusing a tab should automatically activate it
6. Press [Tab] and confirm focus moves to the first tabbable element in the `tabpanel` contents.

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/font-library-modal/index.js
Comment (ciampo) [inline packages/edit-site/src/components/global-styles/font-library-modal/index.js:36]: Let's make sure that removing this classname doesn't cause unexpected consequences, maybe by looking at the PR where it was introduced to see why it was there in the first place

File: packages/edit-site/src/components/global-styles/font-library-modal/index.js
Comment (ciampo) [inline packages/edit-site/src/components/global-styles/font-library-modal/index.js]: Tell me this file doesn't have type checks without telling me that this file doesn't have type checks

```suggestion
								focusable={ false }
```

**Review by ciampo — COMMENTED** [review]
I rebased this PR to include the latest changes (including updates to node/npm version).

> Just for fun, modify the order of the `DEFAULT_TABS` array

I had lots of fun, thank you for recommending this activity!

File: packages/edit-site/src/components/global-styles/font-library-modal/index.js
Comment (chad1008) [reply packages/edit-site/src/components/global-styles/font-library-modal/index.js:36]: Good idea! Looking at [the PR in question](https://github.com/WordPress/gutenberg/pull/54566) I don't see issues. I suspect it was added because `classname` was an optional but very commonly used prop in the `tabs` array of `TabPanel`.

**Comment by chad1008** [comment]
rebased, CI checks are passing!

File: packages/edit-site/src/components/global-styles/font-library-modal/tab-layout.js
Comment (ciampo) [inline packages/edit-site/src/components/global-styles/font-library-modal/tab-layout.js:1]: Nitpicky, but hopefully useful to future devs: let's also rename the file and the React component from "Tab" to "Tabpanel" to better match what the component is used for

**Review by ciampo — APPROVED** [review]
🚀 

Good to merge after all feedback gets addressed

**Comment by chad1008** [comment]
Thanks @ciampo, waiting on CI checks now!

**Pull request merged**
